src/toil/job.py

```diff
@@ -1698,3 +1783,59 @@ def _resolve(cls, jobStoreString, jobStoreFileID):
             # corrupted
             value = cPickle.load(fileHandle)
             return value
+
+
+class PromisedRequirement(object):
+    def __init__(self, *args):
+        """
+        Class for manipulating Promise instances.
+        (see toil.job.Promise)
+
+        :param args[0]: function to act on instance args
+                        default: lambda x: x
+        :param int|Promise args[1:]: list of args, ordered with respect to function parameters
+        """
+        if hasattr(args[0], '__call__'):
+            func = args[0]
+            args = args[1:]
+        else:
+            func = lambda x: x
+
+        self._func = dill.dumps(func)
+        self._args = list(args)
+
+    def get_value(self):
+        """
+        Returns PromisedRequirement value
+        """
+        func = dill.loads(self._func)
+        return func(*self._args)
+
+
+def convertPromiseToPromisedRequirement(kwargs):
```

Comment:
Combine these two methods. It should be possible to convert any promised requirements and then return True if any conversions were made.

Also make the resulting function a static method of the PromisedRequirement class and rename it to just `convertPromises`.